packages/extension-character-count/src/character-count.ts

```diff
@@ -92,6 +98,32 @@ export const CharacterCount = Extension.create<CharacterCountOptions, CharacterC
     }
   },
 
+  onCreate() {
+    const { editor, options } = this
+    const { limit, trimInitialContent } = options
+
+    // Only proceed if a limit is set and trimInitialContent is true
+    if (limit === null || limit === undefined || limit === 0 || !trimInitialContent) { return }
+
+    // Get the current document
+    const doc = editor.state.doc
+
+    // Calculate current character count
+    const currentCharCount = this.storage.characters()
+
+    // If content exceeds limit, trim it
+    if (currentCharCount > limit) {
+      const text = doc.textBetween(0, doc.content.size, undefined, ' ')
+      const trimmedContent = text.slice(0, limit)
+
+      // Replace the entire document content
+      editor.commands.setContent(trimmedContent)
+
+      // Add a console warning
+      console.warn(`[CharacterCount] Initial content exceeded limit of ${limit} characters. Content was automatically trimmed.`)
```

Comment:
I'd prefer to leave this to the plugin, by issuing a transaction or allowing a transaction through which would execute the existing logic.

Prevents drift between the two implementations. Especially since this implementation does not even do the same thing as the existing implementation, you are slicing based on text alone while the existing implementation is slicing the whole range by counted chars